### PR TITLE
HUGE front-end update!

### LINKED ISSUE #47: Ordering the `webapp` directory
Files inside `webapp` should be ordered based upon what their purpose is. Three main directories should be created: `Html`, `CSS` and `JavaScript`.
This way our code should be much easier to read.

### LINKED ISSUE #61: Changes to Navbar
The top navbar must be redesigned and improved with the following changes:
- Add a drop down menu with a list of links to each of the Activities pages, those are:
    - concerti
    - spettacoli teatrali
    - eventi sportivi
    - visite guidate a mostre
    - musei

Special cases:
- If the session attribute `isAdmin` is set, the navbar should also display the links to `Visualizza Utenti` and `Visualizza Dipendenti`
- If the session is set but the `isAdmin` attribute is not set, the navbar should also display a link to `Il Mio Profilo`

### LINKED ISSUE #62: Changes to footer
The footer should include a realistic set of information about the company, such as
- Indirizzo sede legale (with a map view pheraps)
- partita iva
- contatti

### LINKED ISSUE #63: Home Page
The application needs a proper home page, this should include
- name and logo of the service
- a table (or something else) with all the event's categories, with links to the corresponding page of each category
- a carousel with the 3 most clicked events
- a message that updates every 15 seconds with all the sales avaiable

### LINKED ISSUE #64: Login displays error
When the login fails, It should display an error directly in the login page using JSP

### LINKED ISSUE #76: Bootstrap refactor
Concept arts;
Login and register
image

### PR DESCRIPTION
**Pull request by Qw3rty26**
This PR closes #47, closes #76, closes #63, closes #61, closes #64 and closes #62

The `/webapp` directory has been ordered, creating new subdirectories, respectively called:

`/jsp` for jsp files
`/html` for html files
`/css` for css files
`/js` for javascript files

The `/homepage` has been hugely upgraded, with a new carousel

The `/signup` page has now been upgraded as well as `/login`

New `/OK` `/error` pages added to handle errors

New modals added to handle cookies and incorrections in the forms the user sends.

_many, many, MANY other things too small to point at but that put together make the difference!_

`cookies` are now supported frontend, still need to manage them with the backend part though.

### CODE REVIEW

**Comment by San7o** [comment]
There are many problems with this PR, basically the application is broken in every way. You should update the references of those files both on the java classes and in the html, referencing header and footer, and the various scripts. The current state of this PR is not ready for Merge, you just committed to save your work pheraps?

File: src/main/java/Error.java [hunk 1]
```diff
@@ -18,7 +18,7 @@ public void doGet (HttpServletRequest req,
   {
 
     res.setCharacterEncoding("UTF-8");
-    req.getRequestDispatcher("./Errorpage.jsp").include(req, res);
+    req.getRequestDispatcher("./jsp/Errorpage.jsp").include(req, res);
```
Comment (San7o) [inline src/main/java/Error.java]: remove the dot

File: src/main/webapp/jsp/Loginpage.jsp [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+
+                  <div class="d-flex align-items-center mb-3 pb-1">
+                    <i class="fas fa-cubes fa-2x me-3" style="color: #ff6219;"></i>
+                    <span class="h1 fw-bold mb-0">TOMTickets</span>
+                  </div>
+
+                  <h5 class="fw-normal mb-3 pb-3" style="letter-spacing: 1px;">Accedi nel tuo profilo</h5>
+
+                  <div data-mdb-input-init class="form-outline mb-4">
+                    <input type="email" id="email" name="email" class="form-control form-control-lg" required/>
```
Comment (San7o) [inline src/main/webapp/jsp/Loginpage.jsp]: input name here is "username" not "email"

Comment (San7o) [reply src/main/webapp/jsp/Loginpage.jsp]: Please do not commit untested code

Comment (San7o) [reply src/main/webapp/jsp/Loginpage.jsp]: Also input type is "text" not "email" since we can login via username

File: src/main/webapp/jsp/Loginpage.jsp [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+                  </div>
+
+                  <div data-mdb-input-init class="form-outline mb-4">
+                    <input type="password" id="password" name="password" class="form-control form-control-lg" required/>
+                    <label class="form-label" for="password">Password</label>
+                  </div>
+
+		  <div class="d-flex justify-content-center align-items-center">
+                  <div class="form-check form-switch">
+  <input class="form-check-input fs-3" type="checkbox" id="flexSwitchCheckDefault" onchange="makePasswordVisible()" autocomplete="off">
```
Comment (San7o) [inline src/main/webapp/jsp/Loginpage.jsp]: Why is this input inside the form? Remove it

File: src/main/webapp/js/Loginpage.js [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+    var passwordBox = document.getElementById("password");
+    if (passwordBox.type === "password"){
+    	passwordBox.type = "text";
+    }else{
+	passwordBox.type = "password";
+    }
+}
+
+function clearFields(){
+   document.forms["loginForm"]["email"].value = "";
```
Comment (San7o) [inline src/main/webapp/js/Loginpage.js]: "username" instead of "email"

File: src/main/webapp/WEB-INF/jsp/Profile.jsp
Comment (San7o) [inline src/main/webapp/WEB-INF/jsp/Profile.jsp:1]: Change header, now located in "../../jsp/Header.jsp"
same for footer

File: src/main/webapp/jsp/SignupPage.jsp [hunk 5]
```diff
@@ -0,0 +1,124 @@
...
+                  <div class="d-flex align-items-center mb-3 pb-1">
+                    <i class="fas fa-cubes fa-2x me-3" style="color: #ff6219;"></i>
+                    <span class="h2 fw-bold mb-0">TOMTickets</span>
+                  </div>
+
+                  <h5 class="fw-normal mb-3" style="letter-spacing: 1px;">Registra un nuovo profilo</h5>
+                  <div class="row">
+                    <div class="col-md-6 mb-2">
+                      <div data-mdb-input-init class="form-outline">
+                        <input type="text" id="nome" name="nome" class="form-control" />
```
Comment (San7o) [inline src/main/webapp/jsp/SignupPage.jsp]: set this as required

File: src/main/webapp/jsp/SignupPage.jsp [hunk 6]
```diff
@@ -0,0 +1,124 @@
...
+                        <label class="form-label" for="nome">Nome</label>
+                      </div>
+                    </div>
+                    <div class="col-md-6 mb-2">
+                      <div data-mdb-input-init class="form-outline">
+                        <input type="text" id="cognome" name="cognome" class="form-control" />
```
Comment (San7o) [inline src/main/webapp/jsp/SignupPage.jsp]: set this as required

File: src/main/webapp/jsp/SignupPage.jsp [hunk 7]
```diff
@@ -0,0 +1,124 @@
...
+                        <label class="form-label" for="cognome">Cognome</label>
+                      </div>
+                    </div>
+                  </div>
+
+                  <div class="row">
+                    <div class="col-md-6 mb-2">
+                      <div data-mdb-input-init class="form-outline">
+                        <input type="date" id="dataNascita" name="dataNascita" class="form-control" />
```
Comment (San7o) [inline src/main/webapp/jsp/SignupPage.jsp]: set this as required

File: src/main/webapp/jsp/SignupPage.jsp [hunk 8]
```diff
@@ -0,0 +1,124 @@
...
+                        <label class="form-label" for="dataNascita">Data di nascita</label>
+                      </div>
+                    </div>
+                    <div class="col-md-6 mb-2">
+                      <div data-mdb-input-init class="form-outline">
+                        <input type="tel" id="telefono" name="telefono" class="form-control" />
```
Comment (San7o) [inline src/main/webapp/jsp/SignupPage.jsp]: set this as required

File: src/main/webapp/jsp/SignupPage.jsp [hunk 9]
```diff
@@ -0,0 +1,124 @@
...
+                        <label class="form-label" for="telefono">Numero di telefono</label>
+                      </div>
+                    </div>
+                  </div>
+
+                  <div data-mdb-input-init class="form-outline mb-2">
+                    <input type="email" id="email" name="email" class="form-control" />
```
Comment (San7o) [inline src/main/webapp/jsp/SignupPage.jsp]: set this as required

File: src/main/webapp/jsp/SignupPage.jsp [hunk 10]
```diff
@@ -0,0 +1,124 @@
...
+                    <label class="form-label" for="email">Indirizzo Email</label>
+                  </div>
+
+                  <div class="row">
+                    <div data-mdb-input-init class="form-outline mb-2">
+                      <input type="text" id="username" name="username" class="form-control" />
```
Comment (San7o) [inline src/main/webapp/jsp/SignupPage.jsp]: set this as required

File: src/main/webapp/jsp/SignupPage.jsp [hunk 11]
```diff
@@ -0,0 +1,124 @@
...
+                      <label class="form-label" for="username">Username</label>
+                    </div>
+                  </div>
+
+                  <div class="row">
+                    <div class="col-md-6 mb-2">
+                      <div data-mdb-input-init class="form-outline">
+                        <input type="getPasswordState()" id="password" name="password" class="form-control" />
```
Comment (San7o) [inline src/main/webapp/jsp/SignupPage.jsp]: set this as required

File: src/main/webapp/jsp/SignupPage.jsp [hunk 12]
```diff
@@ -0,0 +1,124 @@
...
+                        <label class="form-label" for="password">Password</label>
+                      </div>
+                    </div>
+                    <div class="col-md-6 mb-2">
+                      <div data-mdb-input-init class="form-outline">
+                        <input type="getPasswordState()" id="ripetiPassword" name="ripetiPassword" class="form-control" />
```
Comment (San7o) [inline src/main/webapp/jsp/SignupPage.jsp]: change this as required

Comment (San7o) [reply src/main/webapp/jsp/SignupPage.jsp]: Use "password" instead of "usePasswordState()"

File: src/main/webapp/jsp/SignupPage.jsp (hunk 11 above)
Comment (San7o) [reply src/main/webapp/jsp/SignupPage.jsp]: What the fine is "getPasswordState()"? use "password"
Indeed, getPasswordState() does not work well with the "makePasswordvisible()" script, it has inverted values

**Review by San7o — CHANGES_REQUESTED** [review]
PLEASE test your code before writing a PR

You should also change the path in Profile.java in line 41 to "/WEB-INF/jsp/Profile.jsp"

**Comment by San7o** [comment]
Upsie I noticed the branch name "Currently Working", keep it up!

**Comment by Tech-Matt** [comment]
Dear Team,

I hope this message finds you well. I am writing to discuss some concerns I have about the current state of our codebase. Please understand that this is not a critique of anyone’s abilities or efforts, but rather an attempt to improve the quality of our work and the efficiency of our team.

After reviewing the code, I’ve noticed several areas that could benefit from improvement. The structure and organization of the code, in particular, seem to lack the clarity and consistency we usually strive for. It’s as if the code has been written without a clear plan or understanding of best practices. This lack of structure can make it difficult for others to understand and maintain the code, which can slow down our development process and lead to bugs and errors.

In some ways, it’s reminiscent of how a toddler might approach a complex task - with enthusiasm and creativity, but without the necessary knowledge or experience to achieve the best results. While this comparison might seem harsh, I believe it accurately reflects the current state of our codebase.

To address these issues, I suggest we take the following steps:

**Code Review:** Let’s start by conducting a thorough code review. This will help us identify specific areas that need improvement and develop a plan to address them.
**Coding Standards:** We should establish and adhere to a set of coding standards. This will ensure that our code is consistent, readable, and maintainable.
**Training:** If necessary, we could consider providing additional training or resources to help our team improve their coding skills.
Refactoring: Once we’ve identified the problem areas, we can begin refactoring the code. This will involve rewriting certain sections of the code to improve its structure and efficiency.
**Testing:** Finally, we should implement rigorous testing to ensure that our code is functioning as expected and to catch any potential issues early on.
Remember, the goal here is not to point fingers or assign blame, but to work together to improve the quality of our code and the success of our projects. I believe that with a little effort and collaboration, we can turn this situation around and create a codebase that we can all be proud of.

Thank you for taking the time to read this message. I look forward to discussing these issues further and working together to improve our codebase.

Best regards, TechMatt

**Comment by Tech-Matt** [comment]
I am stepping down as leader of this estabilishment, 
I hope you grow as coders,
Best Regards,

Darth Vader 0_0

**Comment by Qw3rty26** [comment]
> Dear Team,
> 
> I hope this message finds you well. I am writing to discuss some concerns I have about the current state of our codebase. Please understand that this is not a critique of anyone’s abilities or efforts, but rather an attempt to improve the quality of our work and the efficiency of our team.
> 
> After reviewing the code, I’ve noticed several areas that could benefit from improvement. The structure and organization of the code, in particular, seem to lack the clarity and consistency we usually strive for. It’s as if the code has been written without a clear plan or understanding of best practices. This lack of structure can make it difficult for others to understand and maintain the code, which can slow down our development process and lead to bugs and errors.
> 
> In some ways, it’s reminiscent of how a toddler might approach a complex task - with enthusiasm and creativity, but without the necessary knowledge or experience to achieve the best results. While this comparison might seem harsh, I believe it accurately reflects the current state of our codebase.
> 
> To address these issues, I suggest we take the following steps:
> 
> **Code Review:** Let’s start by conducting a thorough code review. This will help us identify specific areas that need improvement and develop a plan to address them. **Coding Standards:** We should establish and adhere to a set of coding standards. This will ensure that our code is consistent, readable, and maintainable. **Training:** If necessary, we could consider providing additional training or resources to help our team improve their coding skills. Refactoring: Once we’ve identified the problem areas, we can begin refactoring the code. This will involve rewriting certain sections of the code to improve its structure and efficiency. **Testing:** Finally, we should implement rigorous testing to ensure that our code is functioning as expected and to catch any potential issues early on. Remember, the goal here is not to point fingers or assign blame, but to work together to improve the quality of our code and the success of our projects. I believe that with a little effort and collaboration, we can turn this situation around and create a codebase that we can all be proud of.
> 
> Thank you for taking the time to read this message. I look forward to discussing these issues further and working together to improve our codebase.
> 
> Best regards, TechMatt

??????

**Comment by San7o** [comment]
> Dear Team,
> 
> I hope this message finds you well. I am writing to discuss some concerns I have about the current state of our codebase. Please understand that this is not a critique of anyone’s abilities or efforts, but rather an attempt to improve the quality of our work and the efficiency of our team.
> 
> After reviewing the code, I’ve noticed several areas that could benefit from improvement. The structure and organization of the code, in particular, seem to lack the clarity and consistency we usually strive for. It’s as if the code has been written without a clear plan or understanding of best practices. This lack of structure can make it difficult for others to understand and maintain the code, which can slow down our development process and lead to bugs and errors.
> 
> In some ways, it’s reminiscent of how a toddler might approach a complex task - with enthusiasm and creativity, but without the necessary knowledge or experience to achieve the best results. While this comparison might seem harsh, I believe it accurately reflects the current state of our codebase.
> 
> To address these issues, I suggest we take the following steps:
> 
> **Code Review:** Let’s start by conducting a thorough code review. This will help us identify specific areas that need improvement and develop a plan to address them. **Coding Standards:** We should establish and adhere to a set of coding standards. This will ensure that our code is consistent, readable, and maintainable. **Training:** If necessary, we could consider providing additional training or resources to help our team improve their coding skills. Refactoring: Once we’ve identified the problem areas, we can begin refactoring the code. This will involve rewriting certain sections of the code to improve its structure and efficiency. **Testing:** Finally, we should implement rigorous testing to ensure that our code is functioning as expected and to catch any potential issues early on. Remember, the goal here is not to point fingers or assign blame, but to work together to improve the quality of our code and the success of our projects. I believe that with a little effort and collaboration, we can turn this situation around and create a codebase that we can all be proud of.
> 
> Thank you for taking the time to read this message. I look forward to discussing these issues further and working together to improve our codebase.
> 
> Best regards, TechMatt

Subject: Re: Re: Concerns Regarding Code Quality

Dear @Tech-Matt ,

I hope this message finds you well, albeit slightly singed from the scathing inferno of your prose. Permit me to illuminate the myriad ways in which your communication has missed the mark, akin to a blindfolded archer aiming for the moon.

Firstly, let's address the tone. Your verbose and flowery language does little to conceal the underlying condescension dripping from every word. Comparing our codebase to the scribbles of a toddler with crayons? Such a metaphor does nothing to foster constructive dialogue; instead, it belittles the efforts of your colleagues and undermines morale.

Moreover, your insistence on grandiloquent imagery only serves to obfuscate the true issues at hand. Rather than engaging in theatrical rhetoric, why not communicate plainly and directly? Clarity, after all, is the cornerstone of effective communication.

Next, let's dissect the substance—or lack thereof—of your proposed solutions. Conducting a code review? Establishing coding standards? Refactoring? These are not revolutionary ideas; they are basic tenets of software development that any competent team should already be employing. Your belabored exposition of these elementary concepts only serves to highlight your own lack of understanding or, worse yet, your assumption of the team's incompetence.

Furthermore, your suggestion of providing additional training comes across as patronizing. Are you implying that your colleagues are not capable of fulfilling their roles without hand-holding? Such insinuations are not only insulting but also indicative of a lack of respect for your peers' capabilities.

In summary, your missive reads less like a sincere attempt to address concerns and more like a self-indulgent exercise in pomposity. If you truly wish to contribute to the improvement of our codebase, I suggest adopting a more humble and collaborative approach—one that respects the contributions of your fellow team members and focuses on tangible solutions rather than theatrical grandstanding.

Regards,

Santo
